src/transformers/modeling_tf_utils.py

```diff
@@ -584,59 +614,121 @@ def __init__(self, config, *inputs, **kwargs):
         self.config = config
         self.name_or_path = config.name_or_path
 
-    def get_input_embeddings(self) -> tf.keras.layers.Layer:
+    def get_input_embeddings(self) -> tf.Variable:
         """
-        Returns the model's input embeddings.
+        Returns the model's input embeddings layer.
 
         Returns:
-            :obj:`tf.keras.layers.Layer`: A torch module mapping vocabulary to hidden states.
+            :obj:`tf.Variable`: The embeddings layer mapping vocabulary to hidden states.
         """
-        base_model = getattr(self, self.base_model_prefix, self)
+        main_layer = getattr(self, self.base_model_prefix, self)
 
-        if base_model is not self:
-            return base_model.get_input_embeddings()
+        if main_layer is not self:
+            try:
+                return main_layer.get_input_embeddings()
+            except AttributeError:
+                self(self.dummy_inputs)
+
+                return main_layer.get_input_embeddings()
         else:
             raise NotImplementedError
 
     def set_input_embeddings(self, value):
         """
-        Set model's input embeddings.
+        Set model's input embeddings
 
         Args:
-            value (:obj:`tf.keras.layers.Layer`):
-                A module mapping vocabulary to hidden states.
+            value (:obj:`tf.Variable`):
+                The new weights mapping hidden states to vocabulary.
         """
-        base_model = getattr(self, self.base_model_prefix, self)
-        if base_model is not self:
-            base_model.set_input_embeddings(value)
-        else:
-            raise NotImplementedError
+        if value is not None:
+            main_layer = getattr(self, self.base_model_prefix, self)
+
+            if main_layer is not self:
+                try:
+                    main_layer.set_input_embeddings(value)
+                except AttributeError:
+                    self(self.dummy_inputs)
+                    main_layer.set_input_embeddings(value)
+            else:
+                raise NotImplementedError
 
-    def get_output_embeddings(self) -> tf.keras.layers.Layer:
+    def get_output_embeddings(self) -> Union[None, tf.Variable]:
         """
         Returns the model's output embeddings
 
         Returns:
-            :obj:`tf.keras.layers.Layer`: A torch module mapping hidden states to vocabulary.
+            :obj:`tf.Variable`: The new weights mapping vocabulary to hidden states.
         """
+
+        if hasattr(self, "get_lm_head"):
+            lm_head = self.get_lm_head()
+            try:
```

Comment:
not a huge fan of this function. It can silently suppress errors and result in some unexpected behavior IMO. I really prefer not to use try....except statements in such a function

Reply:
Why do we need this in the first place?

Reply:
Because we need to be sure that the model is entirely built and not partially.